DataConnectors/SalesforceServiceCloud/SalesforceSentinelConnector/__init__.py

```diff
@@ -0,0 +1,204 @@
+        exit()
+
+
+def generate_date():
+    if interval == 'hourly':
+        current_time = datetime.datetime.utcnow().replace(second=0, microsecond=0)
+        past_time = current_time - datetime.timedelta(hours=hours_interval)
+    elif interval == 'daily':
+        current_time = datetime.datetime.utcnow().replace(hour=0, minute=0, second=0, microsecond=0)
+        past_time = current_time - datetime.timedelta(days=days_interval, hours=1)
+    return past_time.strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+def pull_log_files():
+    past_time = generate_date()
+
+    if interval == 'hourly':
+        query = "/services/data/v44.0/query?q=SELECT+Id+,+EventType+,+Interval+,+LogDate+,+LogFile+,+LogFileLength" + \
+        "+FROM+EventLogFile" + \
+        f"+WHERE+Interval+=+'Hourly'+and+CreatedDate+>+{past_time}"
+
+    elif interval == 'daily':
+        query = "/services/data/v44.0/query?q=SELECT+Id+,+CreatedDate+,+EventType+,+LogDate+,+LogFile+,+LogFileLength" + \
+                "+FROM+EventLogFile" + \
+                f"+WHERE+LogDate+>+{past_time}"
+    try:
+        logging.info('Searching files last modified from {}'.format(past_time))
+        r = requests.get(f'{instance_url}{query}', headers=headers)
+    except Exception as err:
+        logging.error(f'File list getting failed. Exiting program. {err}')
+
+    if r.status_code == 200:
+        files = json.loads(r.text)['records']
+        logging.info('Total number of files is {}.'.format(len(files)))
+        return files
+    else:
+        logging.error(f'File list getting failed. Exiting program. {r.status_code} {r.text}')
+
+
+def get_file_raw_lines(file_url, file_in_tmp_path):
+    url = f'{instance_url}{file_url}'
+    try:
+        with requests.get(url, stream=True, headers=headers) as r:
+            with open(file_in_tmp_path, 'wb') as f:
+                for chunk in r.iter_content(chunk_size=1024*1024):
+                    if chunk:  # filter out keep-alive new chunks
+                        f.write(chunk)
+            if r.status_code == 200:
+                print('File successfully downloaded from url {} '.format(url))
+            else:
+                print('File downloading failed. {r.status_code} {r.text} {file_url}')
+    except Exception as err:
+        print('File downloading failed. {err} {file_url}')
+
+
+def gen_chunks_to_object(file_in_tmp_path, chunksize=100):
+    field_names = [name.lower() for name in list(csv.reader(open(file_in_tmp_path)))[0]]
+    field_names = [x if x != 'type' else 'type_' for x in field_names]
+    reader = csv.DictReader(open(file_in_tmp_path), fieldnames=field_names)
+    chunk = []
+    for index, line in enumerate(reader):
+        if (index % chunksize == 0 and index > 0):
+            yield chunk
+            del chunk[:]
+        chunk.append(line)
+    yield chunk
+
+def gen_chunks(file_in_tmp_path):
+    for chunk in gen_chunks_to_object(file_in_tmp_path, chunksize=10000):
```

Comment:
The above default value is 100. 
Maximum of 30 MB per post to Log Analytics Data Collector API. 
Not sure if the row size can exceed 3KB but possible, 100 is much safer.

Reply:
100 is too low because log files might be quite big (~2Gb) and sending data having chunks of 100 events will take a lot of time. We can safely use  2000. Will change this to 2000.